File: _assets/scripts/normalizers/animation_controllers_fix_old.lua
return function (data)
    local data_attr = python.as_attrgetter(data)
    if data_attr.__contains__("animation_controllers") then
        return
    end
    local output = python.builtins.dict()
    output["defined_in"] = data["defined_in"]
    output["animation_controllers"] = data
    python.as_attrgetter(output["animation_controllers"]).__delitem__("defined_in")
    return output
end

File: _assets/scripts/normalizers/animation_controllers_fix_old.py
from typing import Any, TypedDict

__all__ = ["animation_controllers_fix_old"]

input_typed_dict = TypedDict("input_typed_dict", defined_in=list[str])
output_typed_dict = TypedDict("output_typed_dict", defined_in=list[str], animation_controllers=Any)

def animation_controllers_fix_old(data:input_typed_dict|output_typed_dict) -> output_typed_dict|None:
    if "animation_controllers" in data:
        return None
    output:output_typed_dict = {"defined_in": data["defined_in"], "animation_controllers": data}
    del output["animation_controllers"]["defined_in"]
    return output
